### ISSUE TITLE
Pod Topology Constraints aren't being enforced if nodepools scale from 0

### ISSUE THREAD

**Issue by arodriguezdlc** [Azure/AKS#2389]
**What happened**:

In an AKS cluster with nodepools on different availability zones, we have configured a statefulset with the following pod topology constraint:
```
topologySpreadConstraints:
- maxSkew: 1
  topologyKey: topology.kubernetes.io/zone
  whenUnsatisfiable: DoNotSchedule
  labelSelector:
    matchLabels:
      app: test
```

However, k8s autoscaler and scheduler aren't aware of topology labels of the nodepools when they are scaled from 0 and no nodes were created before. For this reason, all pods are created at the same nodepool in the same zone, that isn't the expected behavior. 

As workaround are creating one node at nodepool creation, although we have the minimun nodes configured to 0, to allow k8s to update its metadata to be aware of the available zones and nodepools. 

**What you expected to happen**:
We expect AKS to be aware of zones and nodepools, even if no nodes of a node pool have been created before. 

**How to reproduce it (as minimally and precisely as possible)**:
- Create an AKS cluster with 3 pools on 3 different ha zones whose minimum number of nodes is 0.
- Create a deployment or statefulset with 3 replicas and a pod topology constraint similar to the shown before.

You will see that all pods are created at the same nodepool and at the same availability zone. 

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): 1.20.7
- Size of cluster (how many worker nodes are in the cluster?): 4
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.): a statefulset with 3 replicas and the pod topology constraint. 
- Others:

**Comment by miwithro** [comment]
@marwanad can you look into this?

**Comment by marwanad** [comment]
I tested the above with the balance-similar-node-groups flag on and  I received 3 nodes in the 3 zonal nodepools.

```
  Normal   TriggeredScaleUp  24s                cluster-autoscaler  pod triggered scale-up: [{aks-pool3-13326359-vmss 0->1 (max: 100)} {aks-pool1-13326359-vmss 0->1 (max: 100)} {aks-pool2-13326359-vmss 0->1 (max: 100)}]
```

Tested on both 1.19 and 1.20. Also, I've noticed that we are missing the population of the topology labels and will push a fix for that with next week's release.

While that theoretically should fix your case, I'm curious why I am not able to repro - are you saying CA fails to scale up any of the pools? Only scales one?

**Comment by marwanad** [comment]
I tested the above with the balance-similar-node-groups flag on and  I received 3 nodes in the 3 zonal nodepools.

```
  Normal   TriggeredScaleUp  24s                cluster-autoscaler  pod triggered scale-up: [{aks-pool3-xxxx-vmss 0->1 (max: 100)} {aks-pool1-xxxx-vmss 0->1 (max: 100)} {aks-pool2-xxxx-vmss 0->1 (max: 100)}]
```

Tested on both 1.19 and 1.20. Also, I've noticed that we are missing the population of the topology labels and will push a fix for that with next week's release.

While that theoretically should fix your case, I'm curious why I am not able to repro - are you saying CA fails to scale up any of the pools? Only scales one?
